app/dependency_source_code_finders/base.rb

```diff
@@ -3,7 +3,7 @@
 
 module DependencySourceCodeFinders
   class Base
-    GITHUB_REGEX = %r{github\.com/(?<repo>[^/]+/[^\./]+)[\./]?}
+    GITHUB_REGEX = %r{github\.com/(?<repo>[^/]+/(?:(?!\.git)[^/])+)[\./]?}
```

Comment:
Yeah, I know. I can't see why it would ever need to get any worse, though.

Alternatives:
- refactor the places where this is used to scrub any `.git` suffixes, but we'd then have half the logic for pulling Github URLs here and half of it elsewhere, which would suck more.
- add a `clean_github_url` method to the base class and call that from the places this regex is used. Better than the above, but feels like overkill